subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/rules/ComponentMetadataRulesSupport.groovy

```diff
@@ -1,39 +0,0 @@
-/*
- * Copyright 2018 the original author or authors.
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *      http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package org.gradle.integtests.resolve.rules
-
-import groovy.transform.CompileStatic
-import org.gradle.api.internal.artifacts.ivyservice.NamespaceId
-
-@CompileStatic
-trait ComponentMetadataRulesSupport {
```

Comment:
In the commit you said that
```
Was only used in IvySpecificComponentMetadataRulesIntegrationTest, and it introduced some weird behavior with Groovy 4.
```

What was the weird behaviour? Is there a Groovy issue for the weird behaviour?

Reply:
I think it was this: https://issues.apache.org/jira/browse/GROOVY-10466

Reply:
Actually, that means we can drop that commit from here.